core/src/main/scala/quasar/datasource/url/Http4sUrlDatasourceModule.scala

```
@@ -245,14 +249,25 @@ class Http4sUrlDatasourceModule(clientBuilder: Http4sClientBuilder)
           case ((_, body), finalize) => body.onFinalize(finalize)
         }
 
+      // we use http-akka because http4s does not parse multiple links
+      // https://github.com/http4s/http4s/issues/2852
       case api @ ApiRequest.HeaderPaginated(req, rel) =>
         resource.allocated map {
           case ((headers, body), finalize) =>
-            val nextUri: Option[Uri] = headers collectFirst {
-              case Link(l) if l.rel.exists(_ === LinkHeaderRelation.name.reverseGet(rel)) =>
-	        l.uri
+            val nextUri: Option[Option[Uri]] = headers collectFirst {
+              case h if h.is(Link) =>
+                HttpHeader.parse("link", h.value) match {
+                  case HttpHeader.ParsingResult.Ok(AkkaLink(links), errors) =>
+                    val relation = LinkParams.rel(LinkHeaderRelation.name.reverseGet(rel))
+                    links collectFirst {
+                      case LinkValue(uri, params) if params.contains(relation) =>
+                        Uri.unsafeFromString(uri.toString)
```

Oh I forgot about this line. We could do the safe thing, and then return `None` if it fails.

---

Though if it does fail right here, that means that either http4s's parsing or akka's rendering of the uri is buggy, and maybe we don't want to swallow that error?

Semantically, returning `None` means that we won't request any more pages, and so the user could unexpectedly not receive all the pages, even if there are more.

---

I agree that we want to propagate the error if this happens. Instead of throwing, I think it'd be safer to raise the exception via `MonadError` so we don't have to rely on `F[_]`'s `map` implementation catching exceptions.